### ISSUE TITLE
FedSGD with virtual batching 

### ISSUE THREAD

**Issue by JohnlNguyen** [facebookresearch/FLSim#43]
## Motivation

Create a memory efficient client to run FedSGD. If a client has many examples, running FedSGD (taking the gradient of the model based on all of the client's data) can lead to OOM. In this PR, we fix this problem by still calling optimizer.step once at the end of local training to simulate the effect of FedSGD.>
